### PR TITLE
Deprecate `is_Map`, `is_...Morphism`

### PR DESCRIPTION
**Pull request by mkoeppe**
### :memo: Checklist

- [x] The title is concise and informative.
- [ ] The description explains in detail what this PR is about.
- [ ] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation and checked the documentation preview.

### CODE REVIEW

File: src/sage/groups/abelian_gps/abelian_group_morphism.py [hunk 1]
```diff
@@ -25,6 +25,8 @@
 
 
 def is_AbelianGroupMorphism(f):
+    from sage.misc.superseded import deprecation
+    deprecation(38103, "The function is_AbelianGroupMorphism is deprecated; use 'isinstance(..., AbelianGroupMorphism)' instead.")
```
Comment (grhkm21) [inline src/sage/groups/abelian_gps/abelian_group_morphism.py]: Isn't this line too long? Why are the linters not triggering 🤔

Comment (mkoeppe) [reply src/sage/groups/abelian_gps/abelian_group_morphism.py]: I've broken some of the long lines in 8619affb424

**Comment by gmou3** [comment]
Looks good. I assume that similar changes need to be applied for `is_Matrix`, etc.

**Comment by mkoeppe** [comment]
> I assume that similar changes need to be applied for `is_Matrix`, etc.

That was done in #38077

**Comment by gmou3** [comment]
Oh, I see; and you did a better job than me on that. Just remove one stray import and good to go as far as I am concerned.

**Comment by mkoeppe** [comment]
Done

**Review by gmou3 — APPROVED** [review]
Thanks.

**Comment by mkoeppe** [comment]
Thanks for the review!

**Pull request merged**